teddy.js

```diff
@@ -542,6 +558,29 @@
         teddy.renderedTemplates[template][l - 1].renderedTemplate = renderedTemplate
       }
 
+      // remove {! comments !} and (optionally) unnecessary whitespace
+      do {
+        oldTemplate = renderedTemplate
+        if (teddy.params.minify) {
+          renderedTemplate = renderedTemplate
+            .replace(/[\f\n\r\t\v]*/g, '')
+            .replace(/\s{2,}/g, ' ')
+        }
+        comments = matchRecursive(renderedTemplate, '{!...!}')
+        l = comments.length
+
+        // only remove comments if noparse tag is not flagged
+        if (dontParse === false || dontParse === undefined) {
+          for (i = 0; i < l; i++) {
+            renderedTemplate = replaceNonRegex(renderedTemplate, '{!' + comments[i] + '!}', '')
+          }
+        } else if (dontParse === true) {
+          renderedTemplate = replaceNonRegex(renderedTemplate, '{!' + comments[0] + '!}', '')
```

Comment:
what's going on here? why remove the first comment? unless i'm missing something, i think the correct logic here would be to do nothing with the comments

Reply:
You're right. This was left over from when I was trying something different.